server/src/main/kotlin/org/kryptonmc/krypton/network/handlers/PlayHandler.kt

```diff
@@ -247,19 +250,28 @@ class PlayHandler(
     }
 
     private fun handlePlayerDigging(packet: PacketInPlayerDigging) {
-        if (player.gameMode !== GameModes.CREATIVE) return
-        if (packet.status != PacketInPlayerDigging.Status.STARTED) return
-
-        val chunkX = packet.location.x() shr 4
-        val chunkZ = packet.location.z() shr 4
-        val chunk = player.world.chunkManager[ChunkPosition.toLong(chunkX, chunkZ)] ?: return
-        val x = packet.location.x()
-        val y = packet.location.y()
-        val z = packet.location.z()
-        chunk.setBlock(x, y, z, Blocks.AIR)
-
-        session.send(PacketOutDiggingResponse(packet.location, 0, PacketInPlayerDigging.Status.FINISHED, true))
-        playerManager.sendToAll(PacketOutBlockChange(Blocks.AIR, packet.location))
+        when(packet.status) {
+            PacketInPlayerDigging.Status.STARTED -> {
+                if (player.gameMode !== GameModes.CREATIVE) return
+                val chunkX = packet.location.x() shr 4
+                val chunkZ = packet.location.z() shr 4
+                val chunk = player.world.chunkManager[ChunkPosition.toLong(chunkX, chunkZ)] ?: return
+                val x = packet.location.x()
+                val y = packet.location.y()
+                val z = packet.location.z()
+                chunk.setBlock(x, y, z, Blocks.AIR)
+
+                session.send(PacketOutDiggingResponse(packet.location, 0, PacketInPlayerDigging.Status.FINISHED, true))
+                playerManager.sendToAll(PacketOutBlockChange(Blocks.AIR, packet.location))
+            }
+            PacketInPlayerDigging.Status.UPDATE_STATE -> {
+                (server.itemManager.handler(player.inventory[player.inventory.heldSlot].type) as KryptonItemTimedHandler).finishUse(player, player.hand)
```

Comment:
We should probably check this here first, especially considering people can override the default handlers for types with their own handlers (which will be addressed later).

```suggestion
val handler = server.itemManager.handler(player.inventory[player.inventory.heldSlot].type)
if (handler !is KryptonItemTimedHandler) return
handler.finishUse(player, player.hand)
```